tiledb/sm/query/query.cc

```diff
@@ -1885,28 +1885,27 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
 Status Query::check_buffers_correctness() {
   // Iterate through each attribute
   for (auto& attr : buffer_names()) {
-    if (buffer(attr).buffer_ == nullptr)
-      return logger_->status(Status::QueryError(
-          std::string("Data buffer is not set for " + attr)));
     if (array_schema_->var_size(attr)) {
       // Var-sized
       // Check for data buffer under buffer_var
-      bool exists_data = buffer(attr).buffer_var_ != nullptr;
-      bool exists_offset = buffer(attr).buffer_ != nullptr;
-      if ((!exists_data && *buffer(attr).buffer_var_size_ != 0) ||
-          !exists_offset) {
+      if (buffer(attr).buffer_var_ == nullptr &&
+          *buffer(attr).buffer_var_size_ != 0) {
+        return logger_->status(Status::QueryError(
+            std::string("Var-Sized input attribute/dimension '") + attr +
+            "' is not set correctly. \nData buffer is not set."));
+      }
+      // Check offset buffer under buffer
+      if (buffer(attr).buffer_ == nullptr) {
```

Comment:
What about the case where the offset buffer is not null, but the data (var) buffer is null?

Reply:
I believe this is covered in the case above (L1891 - 1896). We check if the `buffer_var_` is null but we expected input (`buffer_var_size_ != 0`)

Reply:
@Shelnutt2 reminded me that as per [this PR](https://github.com/TileDB-Inc/TileDB/pull/2481), we allow both null buffers and zero-sized data buffers of var-length.